fs/sync/sync.go

```diff
@@ -1152,6 +1177,9 @@ func (s *syncCopyMove) setDelayedDirModTimes(ctx context.Context) error {
 			}
 			item := item
 			g.Go(func() error {
+				if _, ok := s.modifiedDirs[filepath.ToSlash(item.dir)]; !ok {
```

Comment:
It would be slightly more efficient to do this before `g.Go` so we don't start go routines we don't need to.

Reply:
I was wondering about that, and whether that's outweighed by being able to do it concurrently.

Reply:
Made this change.